memory_backend.go

```diff
@@ -121,7 +123,13 @@ func (m *MemBackend) Listen(ctx context.Context, queue string, h Handler) (err e
 	m.handlers.Store(queue, h)
 	m.queues.Store(queue, make(chan Job, queueCapacity))
 
-	err = m.start(ctx, queue)
+	cctx, cancel := context.WithCancel(ctx)
```

Comment:
Probably no need to create a new context var here, just reassign the `ctx`. When would you want to use the original context without the cancellable?